### PR TITLE
Fix issues with arangodump encryption and compression

### PR DESCRIPTION
**Pull request by danielhlarkin**
### Scope & Purpose

We have three significant issues related to `arangodump`:
1. The code which is supposed to disable compression when encryption is enabled has not been functioning.
2. The code which allows you to disable compression via `--compress-output` has not been functioning.
3. The `dump_encrypted` test suite has not actually enabled compression for `arangodump` and `arangorestore`.

- [x] Bug-Fix for *devel-branch*
- [x] Bug-Fix for a *released version* (backports coming soon)

#### Related Information

- [x] There is a *JIRA Ticket number*: ES-605

### Testing & Verification

Jenkins: https://jenkins.arangodb.biz/job/arangodb-matrix-pr/9957/

### CODE REVIEW

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/9961/

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,8 @@
 devel
 -----
+
+* Fixed issue ES-605: arangodump broken when using encryption
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Fixed issue ES-605: arangodump broken when using encryption.
```

File: arangosh/Utils/ManagedDirectory.cpp [hunk 2]
```diff
@@ -279,16 +278,16 @@ ManagedDirectory::ManagedDirectory(application_features::ApplicationServer& serv
...
-
 #ifdef USE_ENTERPRISE
   ::writeEncryptionFile(_path, _encryptionType, _encryptionFeature);
 #else
   ::writeEncryptionFile(_path, _encryptionType);
 #endif
+
+  // currently gzip and encryption are mutually exclusive, encryption wins
+  if (::EncryptionTypeNone != _encryptionType) {
+    _writeGzip = false;
```
Comment (joerg84) [inline arangosh/Utils/ManagedDirectory.cpp:289]: Should that trigger a log message?

**Comment by joerg84** [comment]
Would prefer some kind of user feedback, but otherwise lgtm

**Pull request merged**